node_groups.tf

```diff
@@ -331,6 +331,8 @@ module "self_managed_node_group" {
 
   for_each = { for k, v in var.self_managed_node_groups : k => v if var.create }
 
+  create = try(each.value.create, true)
```

Comment:
I haven't tested this yet, but any changes we make like this should also be propagated to the EKS managed node group and Fargate profile definitions

Reply:
Good point, I added the feature to the managed node groups and fargate profiles but did not test it, because I am not using them anywhere